.lagoon.yml

```diff
@@ -16,3 +9,73 @@ environments:
         schedule: "0 * * * *"
         command: 'drush cron --root=/app'
         service: cli
+
+# Note regarding tasks in the scaffold.
+# Please avoid making scripts more than one or two lines, and avoid if/else logic.
+# When PaaS users or the support team modify this file, it should be easy to turn things on and off.
+
+tasks:
+  pre-rollout:
+    - run:
+        name: Ensure backup folder exists - not all current envs have one
+        command: mkdir -p /app/web/sites/default/files/private/backups
+        service: cli
+        shell: bash    
+    - run:
+        name: Snapshot the database and store
+        command: if [[ "$LAGOON_ENVIRONMENT_TYPE" = "production" ]]; then drush sql:dump --root=/app --gzip --result-file=/app/web/sites/default/files/private/backups/pre-deploy-dump.sql; fi
```

Comment:
use `SNAPSHOTS` maybe?

Reply:
We have a race condition, it is not initially ready in pre-rollout.

Reply:
Ok, i got the race condition here. Then that variable is obsolete and confusing. Consumers will think that changing it's value will have some effect but in reality it won't.

Reply:
I do not doubt it but we plan to follow up on it.